app/src/sharedTest/java/org/oppia/app/topic/TopicFragmentTest.kt

```diff
@@ -0,0 +1,347 @@
+    onView(
+      allOf(
+        withText("REVIEW"),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+    onView(
+      allOf(
+        atPosition(R.id.review_skill_recycler_view, 0).also { withText("An important skill") },
+        isDescendantOfA(withId(R.id.review_skill_recycler_view))
+      ).also { withId(R.id.skill_name) })
+  }
+
+  @Test
+  @UiThread
+  fun testTopicActivity_showsTopicActivityWithMultipleTabs_clickOnOverviewTab_isTabLayoutDisplayedWithOverviewTabSelected() {
+    activityTestRule.launchActivity(null)
+    onView(
+      allOf(
+        withText("REVIEW"),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+    onView(
+      allOf(
+        withText("OVERVIEW"),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+
+    onView(withId(R.id.topic_tabs_container)).check(matches(matchCurrentTabTitle("OVERVIEW")))
+
+  }
+
+  @Test
+  @UiThread
+  fun testTopicActivity_clickOnOverviewTab_isOverviewTabSelectedAndContentMatched() {
+    activityTestRule.launchActivity(null)
+    onView(
+      allOf(
+        withText("REVIEW"),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+    onView(
+      allOf(
+        withText("OVERVIEW"),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+    onView(withId(R.id.topic_name_text_view)).check(
+      matches(
+        withText(
+          Matchers.containsString(topicName)
+        )
+      )
+    )
+  }
+
+  @Test
+  @UiThread
+  fun testTopicActivity_clickOnPlayTab_configurationChange_isSameTabAndItsDataDisplayed() {
+    activityTestRule.launchActivity(null)
+    onView(
+      allOf(
+        withText("PLAY"),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+    onView(withText("First Story")).check(matches(isDisplayed()))
+    activityTestRule.activity.requestedOrientation = Configuration.ORIENTATION_LANDSCAPE
+    onView(withId(R.id.topic_tabs_container)).check(matches(matchCurrentTabTitle("PLAY")))
+    onView(withText("First Story")).check(matches(isDisplayed()))
+  }
+
+  @Test
+  @UiThread
+  fun testTopicActivity_clickOnTrainTab_configurationChange_isSameTabAndItsDataDisplayed() {
+    activityTestRule.launchActivity(null)
+    onView(
+      allOf(
+        withText("TRAIN"),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+    onView(withText("Master These Skills")).check(matches(isDisplayed()))
+    activityTestRule.activity.requestedOrientation = Configuration.ORIENTATION_LANDSCAPE
+    onView(
+      allOf(
+        withText("TRAIN"),
+        isDisplayed()
+      )
+    )
+    onView(withText("Master These Skills")).check(matches(isDisplayed()))
+  }
+
+  fun testTopicActivity_clickOnReviewTab_configurationChange_isSameTabAndItsDataDisplayed() {
```

Comment:
Is this a test case? Its not annotated with `Test`.

Reply:
Resolved